pkg/prometheus/operator.go

```diff
@@ -1799,6 +1799,12 @@ func (c *Operator) selectServiceMonitors(ctx context.Context, p *monitoringv1.Pr
 				break
 			}
 
+			if endpoint.Interval != "" || endpoint.ScrapeTimeout != "" {
```

Comment:
we may have a situation where `endpoint.ScrapeTimeout` is set, `endpoint.Interval` is empty and the former is greater than the value of `p.Spec.ScrapeInterval` which would cause an error. In short I don't think that we need a test here, `validateScrapeIntervalAndTimeout` should take care of everything.